### ISSUE TITLE
cargo dist fails to compile from source with rust 1.78.0

### ISSUE THREAD

**Issue by rob-p** [axodotdev/cargo-dist#1204]
I get the following error when installing via `cargo install cargo-dist`

```{shell}
error[E0716]: temporary value dropped while borrowed
   --> /home/XXX/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-dist-0.19.0/src/init.rs:882:10
    |
873 |     let install_path = if install_path.is_none()
    |         ------------ borrow later stored here
...
882 |         &Some(InstallPathStrategy::default_list())
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
883 |     } else {
    |     - temporary value is freed at the end of this statement
    |
    = note: consider using a `let` binding to create a longer lived value

For more information about this error, try `rustc --explain E0716`.
error: could not compile `cargo-dist` (lib) due to 1 previous error
error: failed to compile `cargo-dist v0.19.0`, intermediate artifacts can be found at `/mnt/scratch1/XXX/cargo-installyt5WVA`.
```

is 1.78.0 a supported Rust version?

**Comment by Gankra** [comment]
I can reproduce this and yes 1.78.0 *should* be fine. I am both shocked that rustc randomly changed borrow semantics and that we accidentally rammed into it.

**Comment by rob-p** [comment]
Yes; I am also _shocked_ that such a seemingly obviously breaking change was made not on an edition boundary! I don't think it's documented anywhere...

**Comment by Gankra** [comment]
It's not a breaking change, the compiler got strictly better at compiling things in 1.79.0 and all of us updated our toolchains before landing the code change that randomly depended on it.

**Comment by rob-p** [comment]
Perhaps my terminology isn't correct. What I mean is I'm surprised they altered the behavior in such a way not on a major boundary without announcing it.  I realize it's not breaking in the sense that something that *doesn't* compile with 1.78 *will* compile with 1.79 and not vice versa... but still I feel like this is a change worth mentioning (perhaps I missed in the release notes though).

**Comment by Gankra** [comment]
https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#extending-automatic-temporary-lifetime-extension

**Comment by rob-p** [comment]
I was just reading :) — I did indeed miss this as in 1.79 I was pretty much concerned only with [this](https://blog.rust-lang.org/2024/06/13/Rust-1.79.0.html#bounds-in-associated-type-position).

**Comment by Gankra** [comment]
I'll have a patch release out in like 30 mins once tests pass

**Comment by rob-p** [comment]
Awesome; thanks for the super-quick fix!

**Comment by Gankra** [comment]
published
